File: MATERIAL/JAVA/ScormPG0106..java
//revisar esse código

import java.util.Scanner;

public class ScormPG0106 {

    public static void escreval(Object... args) {
        for (Object arg : args) {
            System.out.printf(arg.toString());
        }
        System.out.println();
    }

    public static Object leia(Object dado) {
        Scanner sc = new Scanner(System.in);
        if (dado instanceof Number) {
            dado = sc.nextInt();
        } else {
            dado = sc.nextLine();
        }
        return dado;
    }

    public static void main(String[] args) {
        int topo, n;
        String[] vet = new String[3];
        String livro = "";
        int opcao = -1;
        String data = "", nome = "";

        n = 3;
        topo = 0;

        do {
            for (int k = 0; k < n && opcao != 0; k++) {
                escreval("Nossa locadora de livros");

                escreval("incluir ... 1");
                escreval("Topo...... ", topo);
                if (topo > 0)
                    for (int i = 0; i < topo; i++)
                        escreval(vet[i]);
                escreval("Menu Operacoes");
                escreval("0 - Sair");
                escreval("1 - ALUGAR");
                escreval("Escolha uma opcao ... ");
                opcao = (int) leia(opcao);
                if (opcao == 1) {
                    escreval("Informe o nome do livro ... ");
                    livro = (String) leia(livro);
                    escreval("Informe o nome do cliente ... ");
                    nome = (String) leia(nome);
                    escreval("Data do emprestimo ... ");
                    data = (String) leia(data);

                    if (topo == n) {
                        escreval("Overflow - Inclusao Nao Executada");
                    } else {
                        vet[topo] = livro + " " + nome + " " + data;
                        topo = topo + 1;
                    }
                }
            }

        } while (opcao != 0);
        escreval("Finalizando....");
    }
}

File: MATERIAL/PYTHON/SCORM-PG0106.py
//TESTAR CÓDIGO

def algoritmo_locadora_de_livros():
    topo = 0
    vet = list()
    y = None
    opcao = -1
    k = 0
    data = ''
    nome = ''

    n = 3
    topo = 0
    while opcao != 0:
        print("Nossa locadora de livros")
        while opcao != 0 or k < n:
            print("incluir ... 1")
            print("Topo...... ", topo)
            if topo > 0:
                for i, s in enumerate(vet):
                    print(vet[i])
            print("Menu Operacoes")
            print("0 - Sair")
            print("1 - ALUGAR")
            tmp = input("Escolha uma opcao ... ")
            opcao = int(tmp)
            if opcao == 1:
                v = input("Informe o nome do livro ... ")
                nome = input("Informe o nome do cliente ... ")
                data = input("Data do emprestimo ... ")
                if topo == 3:
                    print(
                        "Overflow - Inclusao Nao Executada - pressione enter")
                    v = input()
                else:
                    topo = topo + 1
                    vet.append(str(v) + ' ' + str(nome) + ' ' + str(data))
        tmp = input("Escolha uma opcao ... ")
        opcao = int(tmp)
    pass


algoritmo_locadora_de_livros()